### PR TITLE
Hamming Exercise

### PR DESCRIPTION
**Pull request by iuliagatea**

### CODE REVIEW

File: Ruby/hamming/hamming.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Hamming
+	def self.compute(s1, s2)
+		raise ArgumentError.new("Sequencies with different length!") if s1.length != s2.length
+		a1 = s1.split("")
+		a2 = s2.split("")
```
Comment (paucio) [inline Ruby/hamming/hamming.rb]: You could set `a1` and `a2` as instance variables in the initializer.

A good structure for single purpose services is:

```
def self.call(params)
    new(params).call
end

def initialize(params)
     @params = params
end

def call
    # the service functionality
end

private

attr_reader :params
```

File: Ruby/hamming/hamming.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Hamming
+	def self.compute(s1, s2)
```
Comment (paucio) [inline Ruby/hamming/hamming.rb]: Try naming the variables with more relevant names... You should make your code easy readable for the reviewer.

Comment (asarbu) [inline Ruby/hamming/hamming.rb]: what change could you make to this code, so that it won't do all the computation in case the sequences are the same ?

Comment (iuliagatea) [reply Ruby/hamming/hamming.rb]: return 0 if s1 == s2 ?

Comment (asarbu) [reply Ruby/hamming/hamming.rb]: 👍

Comment (iuliagatea) [reply Ruby/hamming/hamming.rb]: like sequence1 or first_sequence / sequence2 or second_sequence?

Comment (asarbu) [reply Ruby/hamming/hamming.rb]: I think sequence1/2 is good 🤔 It's important to have relevant names to variables, not just letters

**Comment by asarbu** [comment]
your solution is really good, but it's not very ruby like.

Let's try and make it more ruby-ish. I suggest you look up: `chars`, `zip` and `select` and try to use those in your solution.

File: Ruby/hamming/hamming.rb (hunk 1 above)
Comment (iuliagatea) [reply Ruby/hamming/hamming.rb]: I made changes to this exercise according to your suggestions

File: Ruby/hamming/hamming.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  def initialize(sequence1, sequence2)
+    @sequence1 = sequence1
+    @sequence2 = sequence2
+  end
+
+  def compute
+    raise ArgumentError, "Sequencies with different length!" unless sequence1.length == sequence2.length
+    @distance = 0
+    @distance if sequence1 == sequence2
```
Comment (asarbu) [inline Ruby/hamming/hamming.rb]: this doesn't look right 🤔

File: Ruby/hamming/hamming.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  def initialize(sequence1, sequence2)
+    @sequence1 = sequence1
+    @sequence2 = sequence2
+  end
+
+  def compute
+    raise ArgumentError, "Sequencies with different length!" unless sequence1.length == sequence2.length
+    @distance = 0
```
Comment (asarbu) [inline Ruby/hamming/hamming.rb]: this variable seems to be used mainly in `compare_sequences`, maybe you could define it the scope of that method only

File: Ruby/hamming/hamming.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    @distance if sequence1 == sequence2
+    compare_sequences
+  end
+
+  def compare_sequences
+    sequence1_chars = @sequence1.chars
+    sequence2_chars = @sequence2.chars
+    combined = sequence1_chars.zip(sequence2_chars)
+    combined.select { |k, v| @distance += 1 if k != v }
```
Comment (asarbu) [inline Ruby/hamming/hamming.rb]: what do `k`, `v` represent ?

**Comment by asarbu** [comment]
also, please don't forget to rebase the branch (it's easier at code review) 😉

File: Ruby/hamming/hamming.rb (hunk 5 above)
Comment (iuliagatea) [reply Ruby/hamming/hamming.rb]: key, value normally. but I forgot to change that.

File: Ruby/hamming/hamming.rb (hunk 4 above)
Comment (iuliagatea) [reply Ruby/hamming/hamming.rb]: done, please check

File: Ruby/hamming/hamming.rb (hunk 3 above)
Comment (iuliagatea) [reply Ruby/hamming/hamming.rb]: I changed that part

File: Ruby/hamming/hamming.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    @sequence2 = sequence2
+  end
+
+  def compute
+    raise ArgumentError, "Sequencies with different length!" unless sequence1.length == sequence2.length
+    return 0 if sequence1 == sequence2
+    compare_sequences
+  end
+
+  def compare_sequences
```
Comment (paucio) [inline Ruby/hamming/hamming.rb]: This can be a private method since it's only used in the compute method and there is no need to expose it outside.

Comment (iuliagatea) [reply Ruby/hamming/hamming.rb]: done :)

File: Ruby/hamming/hamming.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  private
+
+  attr_reader :sequence1, :sequence2
+
+  def compare_sequences
+    distance = 0
+    sequence1_chars = @sequence1.chars
+    sequence2_chars = @sequence2.chars
+    combined = sequence1_chars.zip(sequence2_chars)
+    combined.select { |seq1_value, seq2_value| distance += 1 if seq1_value != seq2_value }
```
Comment (asarbu) [inline Ruby/hamming/hamming.rb]: there is a method that you could instead of increasing a counter

Comment (iuliagatea) [reply Ruby/hamming/hamming.rb]: next or succ?  how should i use them? distance = distance.next ?

Comment (asarbu) [reply Ruby/hamming/hamming.rb]: I am referring to method `count`. Using that you don't need the variable at all

Comment (iuliagatea) [reply Ruby/hamming/hamming.rb]: thanks again 😃 Ruby is awesome!
i updated the method

**Pull request merged**